### PR TITLE
fix: delete only unnecessary generated files

### PR DESCRIPTION
**Pull request by mcollovati**
Prevents built tool plugins to to delete all frontend genreated files as first step in the prepare-frontend task, moving the login in BuildFrontendUtil and changing it to only delete files that are no longer created by node tasks.

Part of #18907

### CODE REVIEW

**Comment by mcollovati** [comment]
This change prevents the Vite restarts caused, for example, by the of deletion and creation of files like `vaadin.ts`, but it does not avoid all the typescript errors.
This happens because some preserved files, like `Flow.ts` or `vaadin.ts`, import other sources that will be deleted because `prepare-frontend` does not handle it; examples are `generated-flow-imports.js` and `theme.js`

**Comment by mcollovati** [comment]
File causing typescript errors on deletion
```
src/main/frontend/generated/flow/generated-flow-imports.js
src/main/frontend/generated/theme.js
src/main/frontend/generated/theme.d.ts
src/main/frontend/generated/theme-flowex.generated.js
src/main/frontend/generated/flow/generated-flow-imports.d.ts
src/main/frontend/generated/theme-flowex.components.generated.js
src/main/frontend/generated/theme-flowex.global.generated.js"
src/main/frontend/generated/views.ts
```

If exclude those from deletion, the browser page is reloaded, but I see a JavaScript error in the console and the page is blank.
```
Uncaught TypeError: Cannot read properties of undefined (reading 'push') at copilot-CAmK1KnO.js:4320:20
```

File: flow-server/src/main/java/com/vaadin/flow/server/frontend/TaskRemoveOldFrontendGeneratedFiles.java [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+
+    private Predicate<Path> isKnownUnhandledFile() {
+        Path flowGeneratedImports = FrontendUtils
+                .getFlowGeneratedImports(
+                        frontendGeneratedFolder.getParent().toFile())
+                .toPath().toAbsolutePath();
+        return path -> path.equals(flowGeneratedImports)
+                || path.equals(flowGeneratedImports
+                        .resolveSibling(FrontendUtils.IMPORTS_D_TS_NAME))
+                || path.equals(frontendGeneratedFolder.resolve("views.ts"))
```
Comment (caalador) [inline flow-server/src/main/java/com/vaadin/flow/server/frontend/TaskRemoveOldFrontendGeneratedFiles.java:132]: views.ts is going to be renamed or not generated at all? @mshabarov can you provide insight in to what is done to it.

File: flow-plugins/flow-plugin-base/src/test/java/com/vaadin/flow/plugin/base/BuildFrontendUtilTest.java [hunk 2]
```diff
@@ -255,6 +261,46 @@ public void propagateBuildInfo_existingTokenFileWithSameContent_doesNotWriteToke
...
+        }
+
+        // Run again to verify useless files have been removed
+        BuildFrontendUtil.prepareFrontend(adapter);
+        Set<Path> generatedFiles = Files.walk(frontendGeneratedFolder.toPath())
+                .filter(Files::isRegularFile).collect(Collectors.toSet());
+
+        Assert.assertFalse(
+                "Files not generated by prepare frontend should not be present",
+                generatedFiles.containsAll(additionalFiles));
```
[the remark this answers is not in the document]
Comment (mcollovati) [reply flow-plugins/flow-plugin-base/src/test/java/com/vaadin/flow/plugin/base/BuildFrontendUtilTest.java:298]: Good catch! Thanks

File: flow-server/src/main/java/com/vaadin/flow/server/frontend/AbstractFileGeneratorFallibleCommand.java [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+
+package com.vaadin.flow.server.frontend;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.Collection;
+import java.util.List;
+
+public abstract class AbstractFileGeneratorFallibleCommand
```
[the remark this answers is not in the document]
Comment (mcollovati) [reply flow-server/src/main/java/com/vaadin/flow/server/frontend/AbstractFileGeneratorFallibleCommand.java:34]: added javadoc

File: flow-plugins/flow-plugin-base/src/test/java/com/vaadin/flow/plugin/base/BuildFrontendUtilTest.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (mcollovati) [reply flow-plugins/flow-plugin-base/src/test/java/com/vaadin/flow/plugin/base/BuildFrontendUtilTest.java]: fixed assertion

File: flow-server/src/main/java/com/vaadin/flow/server/frontend/TaskRemoveOldFrontendGeneratedFiles.java (hunk 1 above)
Comment (mshabarov) [reply flow-server/src/main/java/com/vaadin/flow/server/frontend/TaskRemoveOldFrontendGeneratedFiles.java:132]: It has been renamed to [file-routes.ts](https://github.com/vaadin/hilla/pull/2271) and soon will be generated only by Hilla (Flow won't generate a stub of this file).

Comment (mcollovati) [reply flow-server/src/main/java/com/vaadin/flow/server/frontend/TaskRemoveOldFrontendGeneratedFiles.java:132]: Thanks. I'll remove `views.ts` from the check

Comment (mcollovati) [reply flow-server/src/main/java/com/vaadin/flow/server/frontend/TaskRemoveOldFrontendGeneratedFiles.java]: removed

**Review by caalador — DISMISSED** [review]

**Review by caalador — APPROVED** [review]

**Pull request merged**